### PR TITLE
ci: print trial logs in more failure cases

### PR DESCRIPTION
**Pull request by rb-determined-ai**
Previously, if an experiment failed to complete within a timeframe, then
it failed to cancel within a timeframe, the trial logs were never
printed.

There was logic in place to try to avoid printing too many trial logs
when there was a mix of successful and failure trials in the same
experiment.  In practice, the number of e2e tests with multiple trials
is extremely low, but the edge cases where tests failed to show useful
logs was non-zero.  It's much better for developers to just always print
all trial logs when the wait_for_state fails.

## Commentary

[Example](https://app.circleci.com/pipelines/github/determined-ai/determined/17644/workflows/c63cd9c5-b76a-4935-845c-d3a642545066/jobs/540501/steps):

```
tests/experiment/test_tf_estimator.py::test_mnist_estimator_const_parallel[True] starting at 2021-12-16 02:36:34
Waited 60 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 120 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 180 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 240 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 300 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 360 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 420 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 480 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 540 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 600 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 660 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 720 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 780 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 840 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 900 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 960 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1020 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1080 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1140 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1200 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1260 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1320 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1380 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1440 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1500 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1560 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1620 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1680 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 1740 seconds for experiment 2 (currently ACTIVE) to reach COMPLETED
Waited 60 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 120 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 180 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 240 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 300 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 360 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 420 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 480 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 540 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 600 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 660 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 720 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 780 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 840 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 900 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 960 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1020 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1080 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1140 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1200 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1260 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1320 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1380 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1440 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1500 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1560 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1620 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1680 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
Waited 1740 seconds for experiment 2 (currently STOPPING_CANCELED) to reach CANCELED
FAILED
```

This was obviously very unhelpful in explaining what happened.

### CODE REVIEW

File: e2e_tests/tests/experiment/experiment.py [hunk 1]
```diff
@@ -442,25 +439,27 @@ def run_list_cli_tests(experiment_id: int) -> None:
     )
 
 
-def report_failed_experiment(experiment_id: int, state: str) -> None:
-    print(
-        "Experiment {} terminated in {} state unexpectedly!".format(experiment_id, state),
-        file=sys.stderr,
-    )
-
+def report_failed_experiment(experiment_id: int) -> None:
```
Comment (azhou-determined) [inline e2e_tests/tests/experiment/experiment.py:442]: would it be useful to add a target/current state parameters here to only dump the problematic trial's logs? not important since most tests only have 1 trial anyway.

**Review by azhou-determined — APPROVED** [review]
👍

File: e2e_tests/tests/experiment/experiment.py (hunk 1 above)
Comment (rb-determined-ai) [reply e2e_tests/tests/experiment/experiment.py:442]: turns out that all the two places where this is called log that info already, each with useful context about where in the test the failure occurred.
